File: platform/python/tests/fixtures/host_tool_agent.py
"""Test fixture: a Python agent that calls a host tool and streams the result.

No model call — verifies the AAP reverse bridge (host_tool_call/result) and that
``ctx.host_tools`` is populated from ``init``.
"""

from llmtoolforge_agent import run


def on_prompt(ctx):
    ctx.assistant_start()
    names = ",".join(t.get("name", "") for t in ctx.host_tools)
    ctx.assistant_delta(f"tools={names};")
    res = ctx.call_host_tool("echo_host", {"msg": ctx.input})
    ctx.assistant_end(f"result={res['resultText']}")


if __name__ == "__main__":
    run(on_prompt, name="host-tool-agent")

File: platform/node/test/fixtures/host-tool-agent.mjs
/**
 * Test fixture: an agent that calls a host tool and streams the bridged result.
 * No model call — verifies the AAP reverse bridge (host_tool_call/result) and
 * that `ctx.hostTools` is populated from `init`.
 */
import { run } from "../../src/runtime.js";

run({
  name: "host-tool-agent",
  async onPrompt(ctx) {
    ctx.assistantStart();
    // Advertise what the host offered, so the test can assert manifest delivery.
    ctx.assistantDelta(`tools=${ctx.hostTools.map((t) => t.name).join(",")};`);
    const res = await ctx.callHostTool("echo_host", { msg: ctx.input });
    ctx.assistantDelta(`result=${res.resultText}`);
    ctx.assistantEnd(`result=${res.resultText}`);
  },
});
